portfolio/src/main/webapp/script.js

```diff
@@ -160,19 +160,24 @@ function clearComments() {
 }
 /** Gets the login status*/
 function getLoginStatus() {
-  fetch('/login').then((response) => response.text())
-      .then((text) => {
-        return text === 'loggedIn';
-      })
-      .then((loggedIn) => {
-        processLoginStatus(loggedIn);
+  fetch('/login').then((response) => response.json())
+      .then((user) => {
+        processLoginStatus(user);
       });
 }
 /** processes the login status
```

Comment:
nit: this comment is not very useful when the function is already called `processLoginStatus` (similarly for other functions above). It would be more useful to explain what does it mean to process the status - or give a summary of what the function is doing.

Other examples of comments that don't bring anything can be, for example:
```
int itemPrice = 0;  // the price of the item
```

comments should be there to explain what the code can't explain itself, i.e. when it's not possible to make the code self-explanatory. Redundant comments make things worse because they add clutter and they add one more thing that needs to be maintained (if you change the code you'll have to change the comments too).

**I'm OK with leaving the comments as they are** if you prefer. This is mostly for you, so that in the future you can revisit/reuse the code and quickly understand what's going on.